languages/thingtalk/en/stream_tables.genie

```diff
@@ -605,22 +605,24 @@ stream : Ast.Expression = {
 
 with_filtered_table_join : Ast.Expression = {
     !notablejoin {
-    table:with_filtered_table ('with' | 'having' | 'that has' | 'that have' | 'which has' | 'which have') subquery:with_filtered_table
-        => C.addComparisonSubquery(table, subquery, null);
+    table:with_filtered_table ('with' | 'having' | 'that has' | 'that have' | 'which has' | 'which have') subquery:with_filtered_table p:out_param_Any_hidden
+        => C.addComparisonSubquery(table, subquery, p.name);
```

Comment:
We should be able to do this with a regular out_param_Any, using the new syntax for templates. This should be a bit more efficient in memory, and also less confusing.